File: src/main/java/cz/vhromada/catalog/mapper/GenreMapper.java
package cz.vhromada.catalog.mapper;

import cz.vhromada.catalog.entity.Genre;

import org.mapstruct.Mapper;

/**
 * An interface represents mapper for genre.
 *
 * @author Vladimir Hromada
 */
@Mapper
public interface GenreMapper {

    /**
     * Maps entity genre to domain genre.
     *
     * @param source entity genre
     * @return mapped domain genre
     */
    cz.vhromada.catalog.domain.Genre map(Genre source);

    /**
     * Maps domain genre to entity genre.
     *
     * @param source domain genre
     * @return mapped entity genre
     */
    Genre mapBack(cz.vhromada.catalog.domain.Genre source);

}

File: src/main/kotlin/cz/vhromada/catalog/mapper/GenreMapper.kt
package cz.vhromada.catalog.mapper

import cz.vhromada.catalog.entity.Genre
import cz.vhromada.common.mapper.Mapper
import org.springframework.stereotype.Component

/**
 * A class represents mapper for genre.
 *
 * @author Vladimir Hromada
 */
@Component("genreMapper")
class GenreMapper : Mapper<Genre, cz.vhromada.catalog.domain.Genre> {

    override fun map(source: Genre): cz.vhromada.catalog.domain.Genre {
        return cz.vhromada.catalog.domain.Genre(
                id = source.id,
                name = source.name!!,
                position = source.position,
                audit = null)
    }

    override fun mapBack(source: cz.vhromada.catalog.domain.Genre): Genre {
        return Genre(
                id = source.id,
                name = source.name,
                position = source.position)
    }

}
